### ISSUE TITLE
Enabling "Capture OSD" option will enable both "Capture OSD" and "Capture Lua". After recording, EmuHawk outputs 2 movie files, both of which cannot be played back.

### ISSUE THREAD

**Issue by tommai78101** [TASEmulators/BizHawk#3552]
### Summary

As shown here in this GIF:

![GIF](https://user-images.githubusercontent.com/6444116/216872758-44a4fa51-eca0-489b-87c3-74076d8ce71c.gif)

I'm attempting to record on-screen display (OSD) text shown from the RAM Watch window. When about to attempt recording, I noticed that the option to "enable Capture OSD" would enable both the "Capture OSD" and the "Capture Lua" option.

Morilli mentioned this looks wrong. And showed a screenshot of a code snippet suggesting it's wrong:

![image](https://user-images.githubusercontent.com/6444116/216873186-4e556290-a6ca-426f-819d-1e8979f6a8e7.png)

After enabling both options, an attempt to make a small test encoding was made, using the FFmpeg writer, with MP4 default configurations and flags.

It produced 2 movie files, which I named as "test", both of which cannot be played back on a Windows Media Player application. I have attached the 2 movie files here:

test.mp4:
https://user-images.githubusercontent.com/6444116/216873377-d56c7f54-90cf-4d8a-9b39-33bd39cf313a.mp4
test_1.mp4:
https://user-images.githubusercontent.com/6444116/216873378-50e1e34f-c329-457b-92ec-6240a7d79587.mp4

I used the game River City Ransom EX, with a simple RAM Watch properties file (.WCH file, compressed into a ZIP file)
[test_ram_watch.zip](https://github.com/TASEmulators/BizHawk/files/10637381/test_ram_watch.zip)

SHA-1:   64D0B723AE4EB3F3589B535FB935618F2769229F
CRC32:   8686436E

Thus far, only tested in Bizhawk 2.9 RC2.

### Repro
[//]: # "If you can't figure out the list of steps, delete this section and put 'heisenbug' in the summary somewhere. If a Lua script can cause the bug, you can embed that instead (as simple as possible please)."
1. In Bizhawk 2.9 RC2, open EmuHawk.
2. Open the game, and load it up using TAStudio.
3. Open the RAM Watch window, and load the .WCH file into the RAM Watch.
4. In the RAM Watch window, enable Settings -> Display Watches on Screen.
5. In EmuHawk, begin recording a movie starting at frame 0.
6. Begin recording with the default configurations for an MP4 file.
7. Record a couple of frames.
8. Stop the recording.
9. EmuHawk should output 2 MP4 files, one of which is named with a `_1` suffix.
10. Attempt to open any one of the 2 files will result in not being able to play back the movie files.

### Output
No known output exceptions nor error log messages.

### Host env.
[//]: # "List the computers you've found the bug with. If there's a version that doesn't have the bug, please put that in too. Here are some examples:"
- BizHawk 2.9 RC2; Win10 x64 Pro 21H2; Intel/NVIDIA (Only tested with the release candidate 2 version.)

**Comment by Morilli** [comment]
Did a quick test and I can confirm two files get created. The first one seems like an erroneously generated file because it doesn't seem to contain any data, while the second file contains the video as normal.

**Comment by tommai78101** [comment]
@Morilli Were you able to play the second video normally? I couldn't get the second video to play, and instead, it would crash my Windows Media Player application on me.

**Comment by Morilli** [comment]
Yeah you probably need a better video player than windows' default for these video files; it should play fine in your browser though.

**Comment by tommai78101** [comment]
@Morilli Now can confirm the second video can be played back on a newly installed VLC Media Player. It's a shame that the default Windows Media Player can't open the file anymore.

**Comment by YoshiRulz** [comment]
The code is correct, or at least it correctly implements the desired behaviour according to that comment.

**Comment by vadosnaprimer** [comment]
Creating a segment with 0 frames can't be correct. It's taking wrong values when initializing an ffmpeg recording, probably assumes OSD is not captured, and then actually sees that resolution is different and splits.

**Comment by nattthebear** [comment]
It's not desired behavior, but it's also not grossly wrong.  If you blindly stitch together all of the segments, it will work fine; if you ignore the empty segment, it will work fine.

**Comment by Morilli** [comment]
> The code is correct, or at least it correctly implements the desired behaviour according to that comment. (UX needs work, as usual...)

How about something like one option saying `Capture Lua` and the other `Capture Lua + OSD`, and toggling either switches the other off? Sounds kinda intuitive in my head, and shouldn't be confusing to see.
Of course, these toggles could potentially just be options in the recording dialog itself...
